extensions/ql-vscode/src/data-extensions-editor/extension-pack-name.ts

```diff
@@ -0,0 +1,93 @@
+const packNamePartRegex = /[a-z0-9](?:[a-z0-9-]*[a-z0-9])?/;
+const packNameRegex = new RegExp(
+  `^(?<scope>${packNamePartRegex.source})/(?<name>${packNamePartRegex.source})$`,
+);
+const packNameLength = 128;
+
+export interface ExtensionPackName {
+  scope: string;
+  name: string;
+}
+
+export function formatPackName(packName: ExtensionPackName): string {
+  return `${packName.scope}/${packName.name}`;
+}
+
+export function autoNameExtensionPack(
+  name: string,
+  language: string,
+): ExtensionPackName | undefined {
+  let packName = `${name}-${language}`;
+  if (!packName.includes("/")) {
```

Comment:
What if `packName` includes 2 or more slashes?

Reply:
In that case it will be handled by the combination of splitting it by a `/` and taking everything after the first slash as the name: `sanitizedParts.slice(1).join("-")`. This test covers that functionality: https://github.com/github/vscode-codeql/blob/3323fd4e3bd4f2cba5755563d3a74277ab5fcc3d/extensions/ql-vscode/test/unit-tests/data-extensions-editor/extension-pack-name.test.ts#L39-L43